### PR TITLE
Wrappers on model parts that prevent gradient propagation

### PR DESCRIPTION
**Pull request by jlibovicky**
This PR introduces simple wrappers/views on various model part interfaces. It calls `tf.stop_gradient` on outputs of the model parts and provides the same interface as its parent.

### CODE REVIEW

**Comment by jindrahelcl** [comment]
Should we add this to a test? (set initial_variables and freeze encoder, perhaps?)

**Comment by jlibovicky** [comment]
Good idea, I'll do it.

**Comment by jlibovicky** [comment]
@jindrahelcl ping

File: tests/tests_run.sh [hunk 1]
```diff
@@ -7,7 +7,8 @@ export PYTHONFAULTHANDLER=1
 
 bin/neuralmonkey-train tests/vocab.ini
 bin/neuralmonkey-train tests/bahdanau.ini
-bin/neuralmonkey-train tests/bpe.ini
+NEURALMONKEY_STRICT= bin/neuralmonkey-train tests/bpe.ini
+bin/neuralmonkey-train tests/bpe.ini -s 'decoder.encoders=[<encoder_frozen>]'
```
Comment (jindrahelcl) [inline tests/tests_run.sh]: teďkon to nainicializuje enkodér na čisto a nic to s tim nedělá? Nechceme sem ještě přidat `-s 'main.initial_variables=["soubor co by měl existovat z nějakýho z předešlých běhů"]`?
